src/components/select-menu/SelectMenu.spec.tsx

```diff
@@ -0,0 +1,220 @@
+    await select.findByRole('listbox');
+    fireEvent.click(select.getByText('History'));
+    await waitFor(() => expect(select.queryByRole('listbox')).toBeFalsy());
+    expect(select.getByText('History')).toBeTruthy();
+
+    userEvent.click(selectElement);
+    await waitFor(() => expect(select.queryByRole('listbox')).toBeTruthy());
+    expect(
+      within(select.getByRole('option', {selected: true})).getByText('History')
+    ).toBeTruthy();
+  });
+
+  it('can select multiple options in multi select Select', async () => {
+    const select = render(<RenderSelectMenu multiSelect />);
+    const selectElement = select.getByRole('combobox') as HTMLElement;
+
+    expect(selectElement.getAttribute('aria-multiselectable')).toBeTruthy();
+    userEvent.click(selectElement);
+    await select.findByRole('listbox');
+    fireEvent.click(select.getByText('Physics'));
+    fireEvent.click(select.getByText('Science'));
+    userEvent.click(document.body);
+
+    await waitFor(() => expect(select.queryByRole('listbox')).toBeFalsy());
+    expect(select.queryByText('Select...')).toBeFalsy();
+    expect(select.getByText('Physics, Science')).toBeTruthy();
+  });
+
+  it('closes options popup when user clicks outside of it', async () => {
+    const select = render(<RenderSelectMenu />);
+    const selectElement = select.getByRole('combobox') as HTMLElement;
+
+    userEvent.click(selectElement);
+    select.findByRole('listbox');
+    userEvent.click(document.body);
+    await waitFor(() =>
+      expect(selectElement.getAttribute('aria-expanded')).toEqual('false')
+    );
+    await waitFor(() => expect(select.queryByRole('listbox')).toBeFalsy());
+  });
+
+  it('renders with options popup open when select is set as default expanded', async () => {
+    const select = render(<RenderSelectMenu defaultExpanded />);
+
+    select.findByRole('listbox');
+  });
+
+  it('can close default expanded select', async () => {
+    const select = render(<RenderSelectMenu defaultExpanded />);
+
+    userEvent.click(document.body);
+    await waitFor(() => expect(select.queryByRole('listbox')).toBeFalsy());
+  });
+
+  it('can be navigated with a keyboard', async () => {
+    const select = render(<RenderSelectMenu />);
+    const selectElement = select.getByRole('combobox') as HTMLElement;
+
+    selectElement.focus();
+    expect(selectElement).toEqual(document.activeElement);
+    expect(selectElement.getAttribute('aria-expanded')).toBe('false');
+    userEvent.keyboard('{space}');
+
+    await waitFor(() =>
+      expect(selectElement.getAttribute('aria-expanded')).toEqual('true')
+    );
+    const selectedOption1 = select.getByRole('option', {name: 'Physics'});
+
+    expect(selectedOption1).toHaveFocus();
+    expect(selectedOption1).toEqual(document.activeElement);
+    userEvent.keyboard('{space}');
+    expect(selectedOption1.getAttribute('aria-selected')).toEqual('true');
+    await waitFor(() => expect(select.queryByRole('listbox')).toBeFalsy());
```

Comment:
waitFor is used on one expect and missing for the other `expect(selectedOption1.getAt...`, is it on purpose?